packages/carequality-sdk/src/client/carequality.ts

```diff
@@ -0,0 +1,123 @@
+import BadRequestError from "@metriport/core/util/error/bad-request";
+import axios, { AxiosInstance, AxiosResponse, AxiosStatic } from "axios";
+import { Bundle as Stu3Bundle, bundleSchema } from "../models/bundle";
+import { OrganizationList } from "../models/organization";
+
+const DEFAULT_AXIOS_TIMEOUT_SECONDS = 120;
+const MAX_COUNT = 1000;
+const JSON_FORMAT = "_format=json";
+
+export enum APIMode {
+  dev = "dev",
+  staging = "stage",
+}
+
+export type Options = {
+  axios?: AxiosStatic; // Set axios if it fails to load
+  timeout?: number;
+};
+
+/**
+ * Currently operates on FHIR Stu3 format.
+ */
+export class Carequality {
+  static devUrl = "https://dev-dir-ceq.sequoiadns.org/fhir-stu3/1.0.1";
+  static stagingUrl = "https://stage-dir-ceq.sequoiaproject.org/fhir-stu3/1.0.1";
+
+  // NOTE: These URLs can be used if we migrate to FHIR R4 format.
+  // static devUrl = "https://directory.dev.carequality.org/fhir";
+  // static stagingUrl = "https://directory.stage.carequality.org/fhir";
+
+  static ORG_ENDPOINT = "/Organization/";
+  readonly api: AxiosInstance;
+  readonly apiKey: string;
+
+  /**
+   * Creates a new instance of the CommonWell API client pertaining to an
+   * organization to make requests on behalf of.
+   *
+   * @param apiKey          The API key to use for authentication.
+   * @param apiMode         The mode the client will be running.
+   * @param options         Optional parameters
+   * @param options.timeout Connection timeout in milliseconds, default 120 seconds.
+   */
+  constructor(apiKey: string, apiMode: string, options: Options = {}) {
+    this.api = axios.create({
+      timeout: options?.timeout ?? DEFAULT_AXIOS_TIMEOUT_SECONDS * 1_000,
+      baseURL: apiMode === APIMode.dev ? Carequality.devUrl : Carequality.stagingUrl,
```

Comment:
I'd throw as fallback instead of relying on `staging`. This is likely to be an issue when we add `prod`: we can forget to update this and the execution won't perform as expected, we'll have to debug until figure out the SDK is pointing to staging.

Reply:
good point! this couldve been a tricky one to find